tests/test_main.py

```diff
@@ -1,6 +1,116 @@
-# TODO: add tests
-from app.main import hello_world
+import io
+import pytest
 
+from contextlib import redirect_stdout
+from app.main import cache
 
-def test_hello_world():
-    assert hello_world() == "Hello, world!"
+
+def test_cache_1():
```

Comment:
Add a more meaningful names to tests: test_cache_2_different_functions.
Make tests more transparent: test should check some functionality - in this code test_cache_1 and test_cache_2 check the same functionality.

You'd rather create 2 tests:
1. test_cache_single_function - and test, that everything works with caching single function.
2. test_cache_mutliple_function - and test, that even when caching 2-3 functions everything works fine